ukfom/ukf.hpp

```diff
@@ -187,32 +187,21 @@ class ukf {
 	{
 		assert(X.size() == 2 * n + 1);
 
-		lapack::cholesky<n> L(sigma);
+        Eigen::LLT<cov> lltOfSigma(sigma); // compute the Cholesky decomposition of A
+        cov L = lltOfSigma.matrixL(); // retrieve factor L  in the decomposition
 
-		if (!L.isSPD())
-		{
-			std::cerr << std::endl << "sigma is not SPD:" << std::endl
-					  << sigma << std::endl
-					  << "---" << std::endl;
-			Eigen::EigenSolver<cov> eig(sigma);
-			std::cerr << "eigen values: " << eig.eigenvalues().transpose() << std::endl;
-		}
-		
-		assert(L.isSPD());
```

Comment:
Actually, `guaranteeSPD` appears to do nothing (try passing a [1, 0; 0, -1] matrix). If you want it to make a semi-definite matrix, you should use an Eigen-decomposition instead of a singular value decomposition (singular values are always non-negative). But what I wanted to say is that `LLT` requires that the matrix is actually symmetric and positive definite (not just semi-positive definite).
Also, if you make a (relatively expensive) Eigen-decomposition, anyway, you can also use the `operatorSqrt()` which it provides.